fdbclient/NativeAPI.actor.cpp

```diff
@@ -1099,6 +1103,11 @@ Future<Void> DatabaseContext::onMasterProxiesChanged() {
 bool DatabaseContext::sampleReadTags() {
 	return clientInfo->get().transactionTagSampleRate > 0 && deterministicRandom()->random01() <= clientInfo->get().transactionTagSampleRate;
 }
+bool DatabaseContext::sampleOnBytes(uint64_t bytes) {
+	if(bytes >= clientInfo->get().transactionTagSampleBytes) return true;
+	ASSERT(clientInfo->get().transactionTagSampleBytes > 0);
+	return deterministicRandom()->random01() < bytes / clientInfo->get().transactionTagSampleBytes;
```

Comment:
I think you'll need to cast to double to avoid integer truncation here:
```suggestion
	return deterministicRandom()->random01() < ((double)bytes) / clientInfo->get().transactionTagSampleBytes;
```